script/tool/lib/src/gradle_check_command.dart

```diff
@@ -120,6 +130,122 @@ class GradleCheckCommand extends PackageLoopingCommand {
+        r'github.com.*wiki.*Plugins-and-Packages-repository-structure.*gradle-structure');
+    final RegExp keyReadRegex =
+        RegExp('if.*System.getenv.*containsKey.*$keyVariable');
+    final RegExp keyUsedRegex =
+        RegExp('maven.*url.*System.getenv.*$keyVariable.*');
+
+    final bool keyPresent =
+        gradleLines.any((String line) => keyPresentRegex.hasMatch(line));
+    final bool documentationPresent = gradleLines
+        .any((String line) => documentationPresentRegex.hasMatch(line));
+    final bool keyRead =
+        gradleLines.any((String line) => keyReadRegex.hasMatch(line));
+    final bool keyUsed =
+        gradleLines.any((String line) => keyUsedRegex.hasMatch(line));
+
+    if (!documentationPresent) {
+      printError('Does not link artifact hub documentation.');
+    }
+    if (!(keyPresent && keyRead && keyUsed)) {
+      printError('Failed Artifact Hub validation. Include the following in '
+          'example root build.gradle:\n$exampleRootGradleArtifactHubString');
+    }
+
+    return keyPresent && documentationPresent && keyRead && keyUsed;
+  }
+
+  /// Validates the top-level settings.gradle for an example app (e.g.,
+  /// some_package/example/android/settings.gradle).
+  bool _validateExampleTopLevelSettingsGradle(
+      RepositoryPackage package, File gradleSettingsFile) {
+    print('${indentation}Validating '
+        '${getRelativePosixPath(gradleSettingsFile, from: package.directory)}.');
+    final String contents = gradleSettingsFile.readAsStringSync();
+    final List<String> lines = contents.split('\n');
+    // This is tracked as a variable rather than a sequence of &&s so that all
+    // failures are reported at once, not just the first one.
+    bool succeeded = true;
+    if (!_validateArtifactHubSettingsUsage(package, lines)) {
+      succeeded = false;
+    }
+    return succeeded;
+  }
+
+  /// String printed as example of valid example root settings.gradle repository
+  /// configuration that enables artifact hub env variable.
+  @visibleForTesting
+  static String exampleRootSettingsArtifactHubString = '''
+// See https://github.com/flutter/flutter/wiki/Plugins-and-Packages-repository-structure#gradle-structure for more info.
+buildscript {
+  repositories {
+    maven {
+      url "https://plugins.gradle.org/m2/"
+    }
+  }
+  dependencies {
+    classpath "gradle.plugin.com.google.cloud.artifactregistry:artifactregistry-gradle-plugin:2.2.1"
+  }
+}
+apply plugin: "com.google.cloud.artifactregistry.gradle-plugin"
+''';
+
+  /// Validates that [gradleLines] reads and uses a artifiact hub repository
+  /// when ARTIFACT_HUB_REPOSITORY is set.
+  ///
+  /// Required in root gradle file.
+  bool _validateArtifactHubSettingsUsage(
+      RepositoryPackage example, List<String> gradleLines) {
+    final RegExp documentationPresentRegex = RegExp(
+        r'github.com.*wiki.*Plugins-and-Packages-repository-structure.*gradle-structure');
```

Comment:
Same comments in these strings about escaping literal `.`s

Reply:
done